### ISSUE TITLE
onclose handler is called twice for the same websocket

### ISSUE THREAD

**Issue by sfpvk** [CrowCpp/Crow#1138]
Hi
```c++
#include <crow.h>

#include <chrono>
#include <thread>
#include <mutex>
#include <cassert>
using namespace std::literals::chrono_literals;


std::mutex g_ws_mtx;
crow::websocket::connection *g_ws_connection;


void ws_ping_thread_fn()
{
   using namespace std;
   this_thread::sleep_for(10s);
   std::lock_guard lck(g_ws_mtx);
   if (g_ws_connection) {
      cout << "ws_ping_thread_fn::send ping, then close ws,"
         " then sleep for 5s," << endl;
      g_ws_connection->send_ping("");
      g_ws_connection->close();
      this_thread::sleep_for(5s);
      cout << "ws_ping_thread_fn::send ping again" << endl;
      g_ws_connection->send_ping("");
   }
}

int main()
{
   crow::SimpleApp app;

   CROW_ROUTE(app, "/")([] {
      return R"(
         <!DOCTYPE html>
         <html>
         <body>
         <h1>Counter: <span id="c">0</span></h1>
         <script>
         const ws = new WebSocket("ws://" + location.host + "/ws");
         ws.onopen = () => {
            console.log("connected");
         };
         </script>
         </body>
         </html>
      )";
   });

   CROW_WEBSOCKET_ROUTE(app, "/ws")
      .onopen([&](crow::websocket::connection &conn) {
         std::lock_guard lck(g_ws_mtx);
         assert(! g_ws_connection);
         g_ws_connection = &conn;
      })
      .onclose([&](crow::websocket::connection &conn,
            [[maybe_unused]]const std::string& reason, uint16_t) {
         std::cout << "on_close for connection " << &conn << std::endl;
         std::lock_guard lck(g_ws_mtx);
         assert(g_ws_connection);
         g_ws_connection = nullptr;
      });

   std::thread ws_ping_thread(ws_ping_thread_fn);
   ws_ping_thread.detach();
   app.port(8080).run();
}
```
Steps: compile and run this; within 10 seconds load page http://127.0.0.1:8080 in web browser. stdout log:
```
(2026-01-27 03:42:51) [INFO    ] Crow/master server is running at http://0.0.0.0:8080 using 2 threads
(2026-01-27 03:42:51) [INFO    ] Call `app.loglevel(crow::LogLevel::Warning)` to hide Info level logs.
(2026-01-27 03:42:54) [INFO    ] Request: 127.0.0.1:42972 0x5608aab165d0 HTTP/1.1 GET /
(2026-01-27 03:42:54) [INFO    ] Response: 0x5608aab165d0 / 200 0
ws_ping_thread_fn::send ping, then close ws, then sleep for 5s,
on_close for connection 0x7f5fd80038f0
ws_ping_thread_fn::send ping again
on_close for connection 0x7f5fd80038f0
game_server: ../src/main.cpp:61: main()::<lambda(crow::websocket::connection&, const std::string&, uint16_t)>: Assertion `g_ws_connection' failed.
Aborted                    (core dumped) ./game_server
```
If i compile with ASAN or TSAN, then they are not show errors. Under the debugger, I found out that it is safe to call the send/close websocket::connection methods from other threads (because of it is call asio::io_context::post/detach under the hood). And if i use Crow in that way, i can't know exactly when ws connection will be closed, other thread can send something by ws in same time. And i expect, that websocket connection is valid until the onclose handler is ended. But a double onclose handler call suggests that something is going wrong.

PS: if comment out assert in onclose, then stdout log is:
```
(2026-01-27 04:25:24) [INFO    ] Crow/master server is running at http://0.0.0.0:8080 using 2 threads
(2026-01-27 04:25:24) [INFO    ] Call `app.loglevel(crow::LogLevel::Warning)` to hide Info level logs.
(2026-01-27 04:25:27) [INFO    ] Request: 127.0.0.1:43980 0x5608122ca380 HTTP/1.1 GET /
(2026-01-27 04:25:27) [INFO    ] Response: 0x5608122ca380 / 200 0
ws_ping_thread_fn::send ping, then close ws, then sleep for 5s,
on_close for connection 0x7f1ab8003cb0
ws_ping_thread_fn::send ping again
on_close for connection 0x7f1ab8003cb0
on_close for connection 0x7f1ab8003cb0
on_close for connection 0x7f1ab8003cb0
```

**Comment by sfpvk** [comment]
I discovered that the onclose handler is not called twice if close() is not called by me directly, but, for example, when the client drops the connection, or does not respond with "pong" to my "ping" and websocket closed by Crow.

**Comment by gittiver** [comment]
just FYI, nevertheless there are two issues with web sockets, one PR is applied to master (https://github.com/CrowCpp/Crow/pull/1113) the other one is in review https://github.com/CrowCpp/Crow/pull/1136

If that one is finished I will prepare a new release, actually I am a little bit short on time....
